sycl/include/CL/__spirv/spirv_ops.hpp

```diff
@@ -1365,7 +1365,15 @@ __clc_BarrierTestWait(int64_t *state, int64_t arrival) noexcept;
 __SYCL_CONVERGENT__ extern __DPCPP_SYCL_EXTERNAL __SYCL_EXPORT void
 __clc_BarrierArriveAndWait(int64_t *state) noexcept;
 
-#ifdef __SYCL_USE_VARIADIC_SPIRV_OCL_PRINTF__
+#if defined(__SYCL_USE_VARIADIC_SPIRV_OCL_PRINTF__) &&                         \
+    !defined(__INTEL_PREVIEW_BREAKING_CHANGES)
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wpedantic"
```

Comment:
Do we need this because using `#warning` causes some warning for pedantic?

Reply:
We do `-Werror` in the E2E tests to ensure our tests are promptly updated not to use deprecated functionality unless we specifically test for it. In the latter case, which is here, we need to explicitly silent the warning.